src/Materials/babylon.pbrMaterial.ts

```diff
@@ -1269,13 +1281,20 @@
 
                 // Colors
                 this._myScene.ambientColor.multiplyToRef(this.ambientColor, this._globalAmbientColor);
-                
-                // GAMMA CORRECTION.
-                this.convertColorToLinearSpaceToRef(this.reflectivityColor, PBRMaterial._scaledReflectivity);
+
+                if (this.metallic !== undefined || this.roughness !== undefined || this.metallicTexture) {
+                    var metallic = this.metallic === undefined ? 1 : this.metallic;
+                    var roughness = this.roughness === undefined ? 1 : this.roughness;
+                    this._effect.setColor4("vReflectivityColor", new Color3(metallic, roughness), 0);

```

Comment:
this would then be able to be similar from the following block without the creation of an object on each call ?